File: Selenium/Activities/Activity12.java
import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.awt.datatransfer.UnsupportedFlavorException;
import java.io.IOException;
import java.time.Duration;

public class Activity12 {
    public static void main(String[] args) throws InterruptedException, IOException, UnsupportedFlavorException {
        WebDriverManager.firefoxdriver().setup();

        WebDriver driver = new FirefoxDriver();
        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(10));

        driver.get("https://training-support.net/selenium/dynamic-controls");
        System.out.println("Title of Dynamic Controls page: " + driver.getTitle());

        WebElement textbox = driver.findElement(By.id("input-text"));

        System.out.println("Input field is enabled: " + textbox.isEnabled());

        driver.findElement(By.id("toggleInput")).click();

        System.out.println("Input field is enabled: " + textbox.isEnabled());

        driver.close();




    }
}

File: Selenium/Activities/Activity12.py
import win32clipboard
from selenium import webdriver
from selenium.webdriver import ActionChains, Keys
from selenium.webdriver.common.by import By
from selenium.webdriver.firefox.service import Service as FirefoxService
from selenium.webdriver.support import expected_conditions
from selenium.webdriver.support.wait import WebDriverWait
from webdriver_manager.firefox import GeckoDriverManager

service = FirefoxService(GeckoDriverManager().install())

with webdriver.Firefox(service=service) as driver:

    driver.get("https://training-support.net/selenium/dynamic-controls")

    print("Title of Dynamic Controls page: ", driver.title)

    wait = WebDriverWait(driver, 10)

    textbox = driver.find_element(By.ID, "input-text")

    textbox_toggle = driver.find_element(By.ID, "toggleInput")

    print("Text box is enabled: ", textbox.is_enabled())

    textbox_toggle.click()

    print("Text box is enabled: ", textbox.is_enabled())

    textbox.send_keys("Selenium with Python")

    driver.close()

